aten/src/ATen/test/NamedTensor_test.cpp

```diff
@@ -63,4 +63,30 @@ TEST(NamedTensorTest, attachMetadata) {
   tensor.unsafeGetTensorImpl()->set_named_tensor_meta(nullptr);
   ASSERT_FALSE(tensor.is_named());
 }
+
+TEST(NamedTensorTest, internalSetNamesInplace) {
+  auto tensor = at::zeros({3, 2, 5, 7});
+  auto N = Dimname::fromSymbol(Symbol::dimname("N"));
+  auto C = Dimname::fromSymbol(Symbol::dimname("C"));
+  auto H = Dimname::fromSymbol(Symbol::dimname("H"));
+  auto W = Dimname::fromSymbol(Symbol::dimname("W"));
```

Comment:
should there be a Dimname::fromString just to save typing?

Reply:
Maybe. There's a side-effect every time one constructs a Symbol via `Symbol::dimname` (it gets added to the global c++ InternedStrings table) and I didn't want any methods on `Dimname` to have side effects.

Reply:
I'm going to add a helper function to make the typing less in this testing file